rusk-wallet/src/wallet.rs

```diff
@@ -350,18 +306,16 @@ impl<F: SecureWalletFile + Debug> Wallet<F> {
             return Err(Error::Unauthorized);
         }
 
-        let wallet = self.connected_wallet().await?;
-        let ssk_index = addr.index()? as u64;
-        let ssk = self.store.retrieve_ssk(ssk_index).unwrap();
-        let vk = ssk.view_key();
-        let psk = vk.public_spend_key();
+        let index = addr.index()?;
+        let vk = derive_phoenix_vk(&self.seed, index);
+        let pk = derive_phoenix_pk(&self.seed, index);
```

Comment:
the `PhoenixPublicKey` is part of the `Address` and doesn't need to be generated again.